packages/compass-shell/src/components/compass-shell/compass-shell.jsx

```diff
@@ -250,6 +251,7 @@ export default connect(
         state.appRegistry.globalAppRegistry.emit('compass:compass-shell:opened');
       }
     },
-    runtime: state.runtime ? state.runtime.runtime : null
+    runtime: state.runtime ? state.runtime.runtime : null,
+    enableShell: state.enableShell ? state.enableShell : false
   })
 )(CompassShell);
```

Comment:
Another example of a possible simplification – you can avoid having to track `enableShell` in the state fairly directly by adding it to the props:

```suggestion
  })
)(withPreferences(CompassShell, ['enableShell'], React));
```